### ISSUE TITLE
Verify signatures for releases

### ISSUE THREAD

**Issue by laurentsimon** [ossf/scorecard#382]
The [Signed-Releases](https://github.com/ossf/scorecard/blob/main/checks/signed_releases.go) check currently looks for known signature files, but does not verify them.
We should try to verify them with the corresponding key.

Note that for [Signed-Tags](https://github.com/ossf/scorecard/blob/main/checks/signed_tags.go), we rely on github's signature verification. We may also do the verification ourselves for defense in-depth (long-term, low priority).

**Comment by laurentsimon** [comment]
Try to integrate with https://sigstore.dev/

**Comment by dlorenc** [comment]
This might be tricky and encourage an anti-pattern. In general putting the public key right next to the signature on the GitHub release would be problematic and not something we want to encourage, so we'd need another mechanism to find the key.

**Comment by laurentsimon** [comment]
Agreed. I have not looked at how sigstore works to be honest. So you're saying it has no concept of identity. Is there another project trying to tackle this problem?

**Comment by dlorenc** [comment]
> Agreed. I have not looked at how sigstore works to be honest. So you're saying it has no concept of identity. Is there another project trying to tackle this problem?

Ah yeah - with sigstore it would be doable!
